lib/oauth2/access_token.ex

```diff
@@ -109,170 +104,20 @@ defmodule OAuth2.AccessToken do
   ```
   """
   def new(response, client) do
-    {std, other} = Dict.split(response, @standard)
-
-    struct __MODULE__, [
-      access_token:  std["access_token"],
-      refresh_token: std["refresh_token"],
-      expires_at:    (std["expires_in"] || other["expires"]) |> expires_at(),
-      token_type:    std["token_type"] |> normalize_token_type(),
-      other_params:  other,
-      client:        remove_headers(client)]
+    {std, other} = Map.split(response, @standard)
+
+    token =
+      struct(__MODULE__, [
+        access_token:  std["access_token"],
+        refresh_token: std["refresh_token"],
+        expires_at:    (std["expires_in"] || other["expires"]) |> expires_at(),
+        token_type:    std["token_type"] |> normalize_token_type(),
+        other_params:  other
+      ])
+    %{client | headers: [], token: token}
```

Comment:
@hassox this is no different than before. Previously there was a function called `remove_headers(client)` and then the client was set inside the token.

The reasoning behind this is that when you use `OAuth2.Client.get_token` depending on the strategy (i.e. the authorization code flow), the `content-type` is `x-www-form-urlencoded`.

Maybe this could be controlled with an option (`reset_client_headers: true`) passed to `AccessToken.new`?